node/src/components/consensus/highway_core/state.rs

```diff
@@ -566,10 +562,28 @@ impl<C: Context> State<C> {
                 return Err(VoteError::ValueAfterTerminalBlock);
             }
         }
-        // TODO: Validate against LNC.
+        self.validate_lnc(wvote)
+            .map_err(|v| VoteError::LncViolated(self.vote(&v).creator))?;
         Ok(())
     }
 
+    /// Validates whether `wvote` violates the LNC rule.
+    /// Returns the first vote's hash that should have been endorsed but wasn't.
+    ///
+    /// Vote violates LNC rule if it cites naively an equivocation.
+    /// If it cites equivocator then it needs to endorse votes that cite equivocating votes.
+    fn validate_lnc(&self, wvote: &WireVote<C>) -> Result<(), C::Hash> {
```

Comment:
Here's another place we might want to write a test to ensure the business logic is working as intended.

Moreover, I think `Option<C::Hash>` is more appropriate than `Result<(), C::Hash>`

Reply:
> Here's another place we might want to write a test to ensure the business logic is working as intended.

This also depends only on the `state` and `wvote`. It's easy to mock.